cmd/embedded-cluster/reset.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"strings"
+
+	autopilot "github.com/k0sproject/k0s/pkg/apis/autopilot/v1beta2"
+	"github.com/urfave/cli/v2"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/replicatedhq/embedded-cluster/pkg/defaults"
+)
+
+type etcdMembers struct {
+	Members map[string]string `json:"members"`
+}
+
+func getEtcdMemberAddress(hostname string) (string, error) {
+	k0s := defaults.K0sBinaryPath()
+	out, err := exec.Command(k0s, "etcd", "member-list").Output()
+	if err != nil {
+		fmt.Println(err)
+		return "", nil
+	}
+	members := etcdMembers{}
+	json.Unmarshal(out, &members)
+	if _, ok := members.Members[hostname]; ok {
+		url, err := url.Parse(members.Members[hostname])
+		if err != nil {
+			return "", err
+		}
+		return strings.SplitN(url.Host, ":", 2)[0], nil
+	} else {
+		return "", errors.New("unable to find host in etcd members list")
+	}
+}
+
+var resetCommand = &cli.Command{
+	Name:  "reset",
+	Usage: "Reset the node this command is run from",
+	Action: func(c *cli.Context) error {
+		fmt.Println("gathering facts...")
+		k0s := defaults.K0sBinaryPath()
+		binName := defaults.BinaryName()
+		adminConfig, err := exec.Command(k0s, "kubeconfig", "admin").Output()
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+
+		hostname, err := os.Hostname()
+		if err != nil {
+			return nil
+		}
+
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+
+		tempKubeConfig, err := os.CreateTemp("", "*")
+		if err != nil {
+			fmt.Println(err)
+			return nil
+		}
+		defer tempKubeConfig.Close()
+		defer os.Remove(tempKubeConfig.Name())
```

Comment:
I believe this is backwards
deferred functions are Last-In-First-Out so you're removing the file before closing it